websites/M/movie-web/presence.ts

```diff
@@ -0,0 +1,294 @@
+	meta:
+		| {
+				meta: {
+					title: string;
+					id: string;
+					year: string;
+					poster: string;
+					type: "movie";
+				};
+				imdbId: string;
+				tmdbId: string;
+		  }
+		| {
+				meta: {
+					title: string;
+					id: string;
+					year: string;
+					poster: string;
+					type: "series";
+					seasons: {
+						id: string;
+						number: number;
+						title: string;
+					}[];
+					seasonData: {
+						id: string;
+						number: number;
+						title: string;
+						episodes: {
+							id: string;
+							number: number;
+							title: string;
+						}[];
+					};
+				};
+				imdbId: string;
+				tmdbId: string;
+		  };
+	episode?: {
+		episodeId: string;
+		seasonId: string;
+	};
+	progress: {
+		time: number;
+		duration: number;
+	};
+}
+
+presence.on("UpdateData", async () => {
+	const { pathname, href, host } = window.location,
+		subdomain = host.split(".")[0],
+		[
+			showTimestamp,
+			showWatchButton,
+			showProgressBar,
+			barLengthString,
+			barTrack,
+			barFill,
+			showLabel,
+		] = await Promise.all([
+			presence.getSetting<boolean>("timestamp"),
+			presence.getSetting<boolean>("watch"),
+			presence.getSetting<boolean>("progress"),
+			presence.getSetting<string>("barLength"),
+			presence.getSetting<string>("barTrack"),
+			presence.getSetting<string>("barFill"),
+			presence.getSetting<boolean>("showLabel"),
+		]),
+		presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/Ggwzwcv.png",
+		};
+
+	if (subdomain === "dev") {
+		if (pathname === "" || pathname.startsWith("/search"))
+			presenceData.startTimestamp = browsingTimestamp;
+		else if (pathname.startsWith("/media")) {
+			const metaObj = await presence.getPageletiable("meta"),
+				metaData: DevMetaData | undefined = Object.values(metaObj ?? {})[0];
+			if (!metaData) return;
+
+			const { media, state } = metaData;
+
+			presenceData.largeImageKey = media.meta.meta.poster;
+
+			if ((state.progress.time && state.progress.duration) !== 0) {
+				presenceData.state = createProgressBar(
+					state.progress.time,
+					state.progress.duration,
+					{
+						barLengthString,
+						barFill,
+						barTrack,
+						showLabel,
+					}
+				);
+			}
+
+			const title = `${media.meta.meta.title} (${media.meta.meta.year})`;
+			if (media.meta.meta.type === "series") {
+				const episodeData = media.episode,
+					episode = media.meta.meta.seasonData.episodes.find(
+						episode => episode.id === episodeData.episodeId
+					);
+
+				presenceData.details = `S${media.meta.meta.seasonData.number}E${episode.number} — ${title}`;
+			} else presenceData.details = title;
+
+			if (state.mediaPlaying.isFirstLoading) {
+				presenceData.smallImageKey = "https://i.imgur.com/lBCZWIB.gif";
+				presenceData.smallImageText = "Loading";
+			} else if (state.mediaPlaying.isPlaying) {
+				const now = new Date();
+				presenceData.endTimestamp = now.setSeconds(
+					now.getSeconds() + state.progress.duration - state.progress.time
+				);
```

Comment:
Why not use the normal way to get timestamps? (Lmk if this is due to api usage or sum)
![image](https://github.com/PreMiD/Presences/assets/42322979/69cdf574-6435-4459-acab-5e3b983deb6c)

```suggestion
				[, presenceData.endTimestamp] = presence.getTimestampsfromMedia(document.querySelector("video"))
```

Reply:
We expose time and duration in metadata anyway. Can't we go with it?

Reply:
> We expose time and duration in metadata anyway. Can't we go with it?

As far as i know this is prefferd